cmd/bootstrap/transit/cmd/pull.go

```diff
@@ -96,14 +104,30 @@ func pull(cmd *cobra.Command, args []string) {
 
 	// move root checkpoint file if node role is execution
 	if role == flow.RoleExecution {
-		// root.checkpoint is downloaded to <bootstrap folder>/public-root-information after a pull
-		rootCheckpointSrc := filepath.Join(flagBootDir, model.DirnamePublicBootstrap, model.FilenameWALRootCheckpoint)
```

Comment:
instead of `root.checkpoint` file, there can now be `root.checkpoint`, `root.checkpoint.1`, `root.checkpoint.2` etc.